### DISCUSSION TITLE
[@types/qrcode] pollutes global definitions with node specific types

### DISCUSSION THREAD

**Discussion by sylann** [DefinitelyTyped/DefinitelyTyped#67141] [Issues with a @types package]
When a project imports from qrcode while having @types/qrcode installed.

Global functions such as setTimeout suddenly get their type definition from:

```
node_modules/.pnpm/@types+node@16.11.56/node_modules/@types/node/timers.d.ts
```

Instead of from:

```
node_modules/.pnpm/typescript@4.7.4/node_modules/typescript/lib/lib.dom.d.ts
```

Looking into the former, I saw that type definitions for some functions (setTimeout, clearTimeout, setInterval, clearInterval, setImmediate, clearImmediate) are not wrapped in the NodeJS namespace just above.
If I move them manually in the namespace, my problem is solved. Of course, this is not a solution.

Is there an issue on your part?

I see plenty of solutions to avoid this problem, but none of them seem appropriate.
In the mean time I'll make do of course, but I'd like to find a better solution.

---
Tested with 1.5.2 and 1.5.4
Typescript 4.7.4

I have already read some discussions where people have the inverse of my problem in the node realm.

[a comment by an automated account is not shown]

**Comment by jakebailey** [comment]
The package depends on node types via https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/qrcode/index.d.ts#L3 and https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/qrcode/package.json#L9; it's possible that there is a different stream library that could be used in the package to make it not depend on node. I do see that upstream claims to support client-side stuff, so that makes sense.
